ClassCastException: org.codehaus.groovy.runtime.GStringImpl cannot be cast to java.lang.String in Gradle build

When running the Gradle build command using `scip-java index`, an error occurs related to GStringImpl and java.lang.String. The error message is as follows:

`class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String (org.codehaus.groovy.runtime.GStringImpl is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader; java.lang.String is in module java.base of loader 'bootstrap')`
The error occurs when attempting to apply the plugin io.micronaut.build.internal.docs during the project configuration phase. The full stack trace is as follows:

```FAILURE: Build failed with an exception.
An exception occurred applying plugin request [id: 'io.micronaut.build.internal.docs']
> Failed to apply plugin class 'io.micronaut.build.docs.JavadocAggregatorPlugin'.
   > A problem occurred configuring project ':benchmarks'.
      > class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String
```
Steps to Reproduce:
Run the Gradle build with the following command:

`gradlew --no-daemon --init-script /path/to/init-script.gradle -Pkotlin.compiler.execution.strategy=in-process -Dsemanticdb.targetroot=/path/to/semanticdb-targetroot clean scipPrintDependencies scipCompileAll`
The build fails with the error shown above.

Expected Behavior:
The build should complete successfully without the error.

Environment:
Gradle version: 8.14.3
Groovy version: 3.0.24
Kotlin version: 2.0.21
JDK version: JDK-17

Additional Notes:
The error might be caused by an incompatibility between the Groovy and JDK versions, or by unresolved dependencies in the Gradle script.

It appears to be an issue with the conversion of GString to a standard Java String.

Debugged with `--stacktrace` and narrowed it down that issue is coming from `SemanticdbGradlePlugin`:
```
Caused by: java.lang.ClassCastException: class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String (org.codehaus.groovy.runtime.GStringImpl is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @5f8ed237; java.lang.String is in module java.base of loader 'bootstrap')
	at scala.collection.IterableOnceOps.exists(IterableOnce.scala:647)
	at scala.collection.IterableOnceOps.exists$(IterableOnce.scala:644)
	at scala.collection.AbstractIterable.exists(Iterable.scala:935)
	at com.sourcegraph.gradle.semanticdb.SemanticdbGradlePlugin.$anonfun$apply$9(SemanticdbGradlePlugin.scala:172)
	at org.gradle.internal.code.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:124)
	at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:110)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:30)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:27)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:48)
	at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction.execute(DefaultCollectionCallbackActionDecorator.java:107)
	at org.gradle.api.internal.collections.CollectionFilter$1.execute(CollectionFilter.java:62)
	at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:185)
	at org.gradle.api.internal.DelegatingDomainObjectSet.all(DelegatingDomainObjectSet.java:60)
	at com.sourcegraph.gradle.semanticdb.SemanticdbGradlePlugin.$anonfun$apply$1(SemanticdbGradlePlugin.scala:107)
	at org.gradle.internal.code.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:124)
	at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.run(DefaultListenerBuildOperationDecorator.java:173)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:30)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:27)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:67)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:167)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:60)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:48)
	at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction.execute(DefaultListenerBuildOperationDecorator.java:170)
	at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:99)
	at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:87)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:44)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:268)
	at org.gradle.internal.event.BroadcastDispatch$SingletonDispatch.dispatch(BroadcastDispatch.java:170)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:84)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:70)
	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:380)
	at org.gradle.internal.event.BroadcastDispatch$CompositeDispatch.dispatch(BroadcastDispatch.java:272)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:160)
	at org.gradle.internal.event.ListenerBroadcast.dispatch(ListenerBroadcast.java:37)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
	at jdk.proxy1/jdk.proxy1.$Proxy93.afterEvaluate(Unknown Source)
	at org.gradle.configuration.project.LifecycleProjectEvaluator$NotifyAfterEvaluate$1.execute(LifecycleProjectEvaluator.java:247)
	at org.gradle.configuration.project.LifecycleProjectEvaluator$NotifyAfterEvaluate$1.execute(LifecycleProjectEvaluator.java:244)
	at org.gradle.api.internal.project.DefaultProject.stepEvaluationListener(DefaultProject.java:1603)
	at org.gradle.configuration.project.LifecycleProjectEvaluator$NotifyAfterEvaluate.run(LifecycleProjectEvaluator.java:253)
	... 239 more
```

--- a/scip-gradle-plugin/src/main/java/org/scip_code/scip_java/gradle/ScipGradlePlugin.java
+++ b/scip-gradle-plugin/src/main/java/org/scip_code/scip_java/gradle/ScipGradlePlugin.java
@@ -51,25 +51,34 @@ public class ScipGradlePlugin implements Plugin<Project> {
                 task.getOptions().setIncremental(false);
 
                 if (pluginAdded) {
-                  List<String> args = task.getOptions().getCompilerArgs();
+                  // Groovy build scripts can append interpolated strings
+                  // (GString) to compilerArgs, so despite the List<String>
+                  // signature the list may hold non-String elements. Iterate
+                  // through a wildcard reference so elements are not implicitly
+                  // cast to String, which throws ClassCastException (issue
+                  // #801).
+                  List<?> args = task.getOptions().getCompilerArgs();
 
                   // It's important we don't add the plugin configuration more
                   // than once, as javac considers that an error.
                   boolean alreadyAdded =
-                      args.stream().anyMatch(arg -> arg.startsWith("-Xplugin:scip"));
+                      args.stream()
+                          .anyMatch(arg -> String.valueOf(arg).startsWith("-Xplugin:scip"));
                   if (!alreadyAdded) {
                     // The random timestamp ensures the sources are _always_
                     // recompiled, so Gradle doesn't cache any state.
                     // TODO: before this plugin is published to Maven Central, we
                     // will need to revert this change - as it can have
                     // detrimental effect on people's builds.
-                    args.add(
-                        "-Xplugin:scip -targetroot:"
-                            + targetRoot
-                            + " -sourceroot:"
-                            + sourceRoot
-                            + " -randomtimestamp="
-                            + System.nanoTime());
+                    task.getOptions()
+                        .getCompilerArgs()
+                        .add(
+                            "-Xplugin:scip -targetroot:"
+                                + targetRoot
+                                + " -sourceroot:"
+                                + sourceRoot
+                                + " -randomtimestamp="
+                                + System.nanoTime());
                   }
                 }
               });

--- a/scip-java/src/test/kotlin/tests/GradleBuildToolTest.kt
+++ b/scip-java/src/test/kotlin/tests/GradleBuildToolTest.kt
@@ -71,6 +71,17 @@ class GradleBuildToolTest : BuildToolHarness() {
         add(
             checkGradleBuild("build-with-Werror", "gradle/build-with-Werror", expectedScipFiles = 2)
         )
+        // Regression test: Groovy string interpolation puts GString elements
+        // (not java.lang.String) into compilerArgs, which used to crash the
+        // plugin with a ClassCastException.
+        // https://github.com/scip-code/scip-java/issues/801
+        add(
+            checkGradleBuild(
+                "gstring-compiler-args",
+                "gradle/gstring-compiler-args",
+                expectedScipFiles = 1,
+            )
+        )
         add(
             checkGradleBuild(
                 "publishing",

--- a/scip-java/src/test/resources/fixtures/gradle/gstring-compiler-args/build.gradle
+++ b/scip-java/src/test/resources/fixtures/gradle/gstring-compiler-args/build.gradle
@@ -1,0 +1,14 @@
+plugins {
+    id 'java'
+}
+repositories {
+    mavenCentral()
+}
+// Groovy string interpolation produces GString elements, not java.lang.String,
+// so compilerArgs may hold non-String values despite its List<String>
+// signature. Regression test for
+// https://github.com/scip-code/scip-java/issues/801
+def maxErrors = 500
+compileJava {
+    options.compilerArgs << "-Xmaxerrs" << "${maxErrors}"
+}

--- a/scip-java/src/test/resources/fixtures/gradle/gstring-compiler-args/src/main/java/example/Example.java
+++ b/scip-java/src/test/resources/fixtures/gradle/gstring-compiler-args/src/main/java/example/Example.java
@@ -1,0 +1,7 @@
+package example;
+
+public class Example {
+    public String greet(String name) {
+        return "Hello, " + name + "!";
+    }
+}

